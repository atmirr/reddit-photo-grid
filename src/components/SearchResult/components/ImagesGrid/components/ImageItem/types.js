// @flow
export type ImageItemProp = {
  title: string,
  thumbnail: string,
};
