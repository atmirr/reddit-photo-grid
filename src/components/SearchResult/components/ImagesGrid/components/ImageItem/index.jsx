// @flow
import React from 'react';
import type { Element } from 'react';
import styled from 'styled-components';
import type { ImageItemProp } from './types.js';

const Item = styled.div`
  padding: 1.5em 1.3em;
  text-align: center;
`;
const ImageWrapper = styled.div`
  display: flex;
  border: 1px solid #dbdbdb;
  border-radius: 2px;
  box-shadow: 0 0 6px 4px #e3e3e3;
  & img {
    border-radius: 2px;
  }
`;

function ImageItem({ title, thumbnail }: ImageItemProp): Element<'div'> {
  return (
    <Item>
      <ImageWrapper>
        <img src={thumbnail} alt={title} width={200} height={200} />
      </ImageWrapper>
    </Item>
  );
}

export default ImageItem;
