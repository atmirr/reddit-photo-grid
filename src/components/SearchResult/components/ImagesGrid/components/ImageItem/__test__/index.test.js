import React from 'react';
import { render } from '@testing-library/react';
import ImageItem from '../index';

const props = {
  title: 'Sample post',
  thumbnail:
    'https://b.thumbs.redditmedia.com/lkIqoXa0Yhxwhy1rjgw0m6dJFnLVyPcZGpj0wzyTW9w.jpg',
};

describe('<ImageItem />', () => {
  it('should renders the component', () => {
    const { container } = render(<ImageItem {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should display an image when thumbnail prop has been set', () => {
    const { container } = render(<ImageItem thumbnail={props.thumbnail} />);
    expect(container.querySelectorAll('img').length).toBeTruthy();
  });

  it("should contain image's alt text when title prop has been set", () => {
    const { queryByAltText } = render(<ImageItem title={props.title} />);
    expect(queryByAltText(props.title)).toBeTruthy();
  });
});
