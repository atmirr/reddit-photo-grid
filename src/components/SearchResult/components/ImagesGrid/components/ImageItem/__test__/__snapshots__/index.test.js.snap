// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageItem /> should renders the component 1`] = `
<div
  class="sc-bdfBwQ gTWBiH"
>
  <div
    class="sc-gsTCUz fGxiCj"
  >
    <img
      alt="Sample post"
      height="200"
      src="https://b.thumbs.redditmedia.com/lkIqoXa0Yhxwhy1rjgw0m6dJFnLVyPcZGpj0wzyTW9w.jpg"
      width="200"
    />
  </div>
</div>
`;
